File: ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/EstadiaHotel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ReservaDeHotel.Models
{
    public class EstadiaHotel
    {
        [Key]
        public int IdEstadia { get; set; }
        public int QtdQuartos { get; set; }
        public DateTime DataEntrada { get; set; }
        public DateTime DataSaida { get; set; }

        public EstadiaHotel()
        {
        }

        public EstadiaHotel(int qtdQuartos, DateTime dataEntrada, DateTime dataSaida)
        {
            QtdQuartos = qtdQuartos;
            DataEntrada = dataEntrada;
            DataSaida = dataSaida;
        }
    }
}

File: ReservaHotelFront/src/app/EstadiaHotel.ts
export class EstadiaHotel {
    IdEstadia: number = 0;
    QtdQuartos: number = 0;
    DataEntrada: Date = new Date();
    DataSaida: Date = new Date();

    constructor(qtdQuartos: number, dataEntrada: Date, dataSaida: Date) {
        this.QtdQuartos = qtdQuartos;
        this.DataEntrada = dataEntrada;
        this.DataSaida = dataSaida;
    }
}
